### PR TITLE
Fix Exception While Creating an Order in the Admin

### PR DESCRIPTION
**Pull request by justin-at-bounteous**
Changes:
- $total->getDiscountDescription() returns a Phrase, but strlen expects a string
- $total->getDiscountDescription()->getText() will return a string

### CODE REVIEW

File: app/code/Magento/SalesRule/Model/Quote/Address/Total/ShippingDiscount.php [hunk 1]
```diff
@@ -89,7 +89,7 @@ public function fetch(Quote $quote, Total $total): array
         $amount = $total->getDiscountAmount();
 
         if ($amount != 0) {
-            $description = $total->getDiscountDescription() ?: '';
+            $description = $total->getDiscountDescription() ? $total->getDiscountDescription()->getText() : '';
```
Comment (orlangur) [inline app/code/Magento/SalesRule/Model/Quote/Address/Total/ShippingDiscount.php]:
```suggestion
            $description = (string)$total->getDiscountDescription() ?: '';
```

**Review by orlangur — CHANGES_REQUESTED** [review]
After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉

File: app/code/Magento/SalesRule/Model/Quote/Address/Total/ShippingDiscount.php (hunk 1 above)
Comment (justin-at-bounteous) [reply app/code/Magento/SalesRule/Model/Quote/Address/Total/ShippingDiscount.php]: Changes implemented. Thanks.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-5034](https://jira.corp.magento.com/browse/ENGCOM-5034) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**